.meta/sinks/gcp_stackdriver_logging.toml

```diff
@@ -0,0 +1,97 @@
+[sinks.gcp_stackdriver_logging]
+title = "GCP Stackdriver Logging"
+batch_max_size = 5242880
+batch_timeout_secs = 1
+beta = true
+buffer = true
+common = true
+delivery_guarantee = "best_effort"
+function_category = "transmit"
+healthcheck = true
+egress_method = "batching"
+input_types = ["log"]
+request_rate_limit_duration_secs = 1
+request_rate_limit_num = 100
+request_retry_attempts = 9223372036854775807
+request_retry_initial_backoff_secs = 1
+request_retry_max_duration_secs = 10
+request_in_flight_limit = 5
+request_timeout_secs = 60
+service_providers = ["Google"]
+write_to_description = "[Google Cloud Platform's Stackdriver Logging service][urls.gcp_stackdriver_logging] via the [REST Interface][urls.gcp_stackdriver_logging_rest]"
+tls_options = []
+
+[sinks.gcp_stackdriver_logging.options.credentials_path]
+type = "string"
+required = true
+description = """\
+The filename for a Google Cloud service account credentials JSON file \
+used to authenticate access to the stackdriver logging API. \
+"""
+
+[sinks.gcp_stackdriver_logging.options.project_id]
+type = "string"
+common = false
+required = true
+examples = ["vector-123456"]
+description = "The project name to which to publish logs."
+
+[sinks.gcp_stackdriver_logging.options.billing_account_id]
+type = "string"
+common = false
+examples = ["vector-123456"]
+description = """\
+The billing account ID to which to publish logs.
+Exactly one of `billing_account_id`, `folder_id`, `log_id`, or `organization_id` must be set.\
```

Comment:
To clarify, a user _cannot_ supply multiple of these attributes, they _must_ supply exactly one? If that's the case I wonder if it would make more sense to have a single polymorphic attribute:

```toml
[sinks.stackdriver]
  type = "gcp_stackdriver_logging"
  billing.type = "account"
  billing.id = "12345"
```

Just food for thought. I don't completely understand what these attributes are used for.

Reply:
The log name is one of `projects/[PROJECT_ID]/logs/[LOG_ID]`, `folders/[FOLDER_ID]/logs[LOG_ID]`, etc. Exactly one of the four attributes must be supplied to construct the log name.

I see however that I made a mistake in which four they are, should include `project_id` instead of `log_id`.